subiquity/controllers/mirror.py

```diff
@@ -38,20 +39,41 @@ class CheckState(enum.IntEnum):
 
 class MirrorController(SubiquityController):
 
+    autoinstall_key = "apt"
     model_name = "mirror"
     signals = [
         ('snapd-network-change', 'snapd_network_changed'),
     ]
 
     def __init__(self, app):
+        self.ai_data = {}
         super().__init__(app)
         self.check_state = CheckState.NOT_STARTED
         if 'country-code' in self.answers:
             self.check_state = CheckState.DONE
             self.model.set_country(self.answers['country-code'])
         self.lookup_task = SingleInstanceTask(self.lookup)
 
+    def load_autoinstall_data(self, data):
+        if data is None:
+            return
+        self.ai_data = data
+        if 'mirror' in data:
```

Comment:
This check fails if users don't provide 'mirror' info in the autoinstall, as the loader sets data['mirror'] = None.
Either the loader shouldn't set default value to None so you can do key checks like this, or this needs to be
if data['mirror'] is not None:

Reply:
I don't understand your comment but I don't think it is correct.

This all needs to change anyway because I want to redo how mirrors are described, see https://github.com/CanonicalLtd/subiquity/pull/626 and https://wiki.ubuntu.com/FoundationsTeam/AutomatedServerInstalls/ConfigReference?action=diff&rev2=40&rev1=38